src/datasources/email/email.datasource.ts

```diff
@@ -8,6 +8,18 @@ import { Email } from '@/datasources/email/entities/email.entity';
 export class EmailDataSource implements IEmailDataSource {
   constructor(@Inject('DB_INSTANCE') private readonly sql: postgres.Sql) {}
 
+  async getVerifiedSignerEmailsBySafeAddress(args: {
+    chainId: string;
+    safeAddress: string;
+  }): Promise<{ email: string }[]> {
+    const emails = await this.sql<Email[]>`
+      SELECT email_address
+      FROM emails.signer_emails
+      WHERE chain_id = ${args.chainId} AND safe_address = ${args.safeAddress} AND verified is true`;
+
+    return emails.map((email) => ({ email: email.email_address }));
```

Comment:
Wouldn't the `SELECT` statement return just email addresses (so a `string`)?

Reply:
Ok nevermind. It returns the column name as a key 👍 

However beware that the type won't match `Email` completely. Instead it will be `{ email_address: string }`

Reply:
I think here it's probably better to use

```typescript
await this.sql<{ email_address: string }[]>``
```

or even

```typescript
await this.sql``
```

So that the rigth type expectations are set.

Reply:
Yeah, good point, fixed in [f1ff610](https://github.com/safe-global/safe-client-gateway/pull/877/commits/f1ff61004d4b054ac07ac7b5750cec600b9d5d8a)